src/main/scala/Refunder.scala

```diff
@@ -1,82 +1,82 @@
 import java.time.LocalDate
 import ZuoraRefundService.{AccountSummary, Invoice, Payment}
-import scalaz.{\/, \/-}
+import scalaz.{-\/, \/, \/-}
 import scalaz.Scalaz._
 
 object Refunder extends App with Logging {
 
   case class AccountForRefund(accountId: String, totalOverpaymentAmount: Double)
 
-  case class OutputCsvLine(accountId: String, negativeInvoiceId: String, negativeInvoiceAmount: Double, invoiceToCreditId: String)
+  case class OutputCsvLine(accountId: String, negativeInvoiceId: String, negativeInvoiceAmount: Double)
 
-  def identifyValidPayment(accountSummary: AccountSummary, totalOverpaymentAmount: Double): String \/ Payment = {
-    val cutOffDate = LocalDate.of(2017, 8, 13)
-    val payments = accountSummary.payments
-    val validPayments = payments.filter(payment => payment.effectiveDate.isAfter(cutOffDate) && payment.status == "Processed" && payment.paidInvoices.head.appliedPaymentAmount > totalOverpaymentAmount)
-    if (validPayments.size == 1) {
-      val validPayment = validPayments.head
-      logInfo(accountSummary.basicInfo.id, s"Found valid payment for refund invoice: $validPayment")
-      validPayment.right
+  def identifyEligiblePayments(accountSummary: AccountSummary, totalOverpaymentAmount: Double): String \/ List[Payment] = {
+    val cutOffDate = LocalDate.of(2017, 10, 31)
```

Comment:
Is there a way to parameterise this date to the command-line args? If someone refuses this script they might not find this line to tweak the date.

Reply:
Good idea - I'll tweak this later